internal/services/costmanagement/view_subscription_resource_test.go

```diff
@@ -0,0 +1,254 @@
+	})
+}
+
+func TestAccSubscriptionCostManagementView_requiresImport(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_subscription_cost_management_view", "test")
+	r := SubscriptionCostManagementView{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		{
+			Config:      r.requiresImport(data),
+			ExpectError: acceptance.RequiresImportError("azurerm_subscription_cost_management_view"),
+		},
+	})
+}
+
+func (t SubscriptionCostManagementView) Exists(ctx context.Context, clients *clients.Client, state *pluginsdk.InstanceState) (*bool, error) {
+	id, err := views.ParseScopedViewID(state.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := clients.CostManagement.ViewsClient.GetByScope(ctx, *id)
+	if err != nil {
+		return nil, fmt.Errorf("retrieving (%s): %+v", *id, err)
+	}
+
+	return utils.Bool(resp.Model != nil), nil
+}
+
+func (SubscriptionCostManagementView) basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+data "azurerm_subscription" "test" {}
+
+resource "azurerm_subscription_cost_management_view" "test" {
+  name            = "testcostview%s"
+  subscription_id = data.azurerm_subscription.test.id
+  chart_type      = "StackedColumn"
+  display_name    = "Test View %s"
+
+  accumulated = "false"
+  query {
+    type      = "Usage"
+    timeframe = "MonthToDate"
+
+    dataset {
+      granularity = "Monthly"
+      sorting {
+        direction = "Ascending"
+        name      = "BillingMonth"
+      }
+      grouping {
+        name = "ResourceGroupName"
+        type = "Dimension"
+      }
+      aggregation {
+        name        = "totalCost"
+        column_name = "Cost"
+      }
+      aggregation {
+        name        = "totalCostUSD"
+        column_name = "CostUSD"
+      }
+    }
+  }
+  kpi {
+    enabled = true
+    type    = "Forecast"
+  }
+  pivot {
+    type = "Dimension"
+    name = "ServiceName"
+  }
+  pivot {
+    type = "Dimension"
+    name = "ResourceLocation"
+  }
+  pivot {
+    type = "Dimension"
+    name = "ResourceGroupName"
+  }
+}
+`, data.RandomString, data.RandomString)
+}
+
+func (SubscriptionCostManagementView) update(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+data "azurerm_subscription" "test" {}
+
+resource "azurerm_subscription_cost_management_view" "test" {
+  name            = "testcostview%s"
+  subscription_id = data.azurerm_subscription.test.id
+  chart_type      = "Line"
+  display_name    = "Test View 2 %s"
+
+  accumulated = "false"
+  query {
+    type      = "Usage"
+    timeframe = "YearToDate"
+
+    dataset {
+      granularity = "Daily"
+      aggregation {
+        name        = "totalCostUSD"
+        column_name = "CostUSD"
+      }
+    }
+  }
+  kpi {
+    enabled = true
+    type    = "Forecast"
+  }
```

Comment:
We typically don't expose the [enabled field for blocks](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/topics/schema-design-considerations.md#features-that-are-toggled-by-the-property-enabled).

Would this potentially be nicer as a list e.g.
1. Enabling `Forecast` -> `kpi = ["Forecast"]`
2. Enabling  `Forecast` and `Budget` -> `kpi = ["Forecast", "Budget"]`
3. Disabling `Forecast` -> `kpi = ["Budget"]`

Reply:
KPI has also an ID property (which I didn't expose yet for simplicity sake), but I'll remove the `enabled` field!